File: Testbed/LifeCycle/examples/fork_counter.pl
#!/usr/bin/perl -w
use strict;
use JSON::XS;
use Getopt::Long;
use Clone qw(clone);

my ($in,$out,$json,$workflow,$payload);
my ($i,@p,$p);
$payload = {};
GetOptions(
                'in=s'  => \$in,
                'out=s' => \$out,
          );

open IN, "<$in" or die "open input $in: $!\n";
$json = <IN>;
close IN;
$payload = decode_json($json);
$workflow = $payload->{workflow};

@p=();
foreach $i (1000, 2000, 3000) {
  $p = clone $payload;
  $p->{workflow}{counter} = $i;
  $p->{workflow}{Intervals}{counter} = 2 * int($i/1000);
  push @p,$p;
  print "fork_counter: create new workflow with counter=$i\n";
}

open  OUT, ">$out" or die "open output $out: $!\n";
print OUT encode_json(\@p);
close OUT;
exit 0;

File: Testbed/LifeCycle/examples/fork_counter.py
#!/usr/bin/env python
import traceback, getopt, sys
import json, copy

try:
  opts, args = getopt.getopt(sys.argv[1:], "io", ["in=", "out="])
except getopt.GetoptError, err:
  print "Error parsing arguments"
  sys.exit(2)
_out = 'none'
_in  = 'none'
for o, a in opts:
  if o == "--out":
    _out = a
  elif o == "--in":
    _in = a
  else:
    assert False, "unhandled option"

f = open(_in,'r')
payload = json.loads(f.read())

payloads=[]
for i in [ 1000, 2000, 3000 ]:
  p = copy.deepcopy(payload)
  p['workflow']['Intervals']['counter'] = 2 * i / 1000
  p['workflow']['counter'] = i
  print "fork_counter: create new workflow with counter=",i
  payloads.append(p)

f = open(_out,'w')
f.write(json.dumps(payloads))
